### DISCUSSION TITLE
UDP broadcast?

### DISCUSSION THREAD

**Discussion by hughfe** [jvde-github/AIS-catcher#136] [General]
Is the idea of accepting a UDP broadcast address under consideration?  I thought this was working at one point, but when I run ais-catcher as a service it does not broadcast.

**Comment by jvde-github** [comment]
Never considered it, but my understanding is that the following broadcast:
```
AIS-catcher -u 192.168.1.255 12345
```
will work on Windows but not great on Linux without setting the SO_BROADCAST option on the socket. That seems fairly simple to do so uploaded an experimental fix that does just that with the following spec:
```
AIS-catcher -u 192.168.1.255 12345 BROADCAST ON
```
Default is `BROADCAST OFF`. Perhaps this already solves your request?

**Reply by hughfe** [reply]
Thank you, that seems to have done the trick.
I wanted to send AIS to my PC for testing without having to check the DHCP assigned address.

**Comment by jvde-github** [comment]
Great! It is an extra layer of security of the OS to require this flag to be set if you want to broadcast. 

Glad that it works, I will keep it in and will consider making this a default setting. Will close this
